Source/NSubstitute/SubstituteExtensions.cs

```diff
@@ -142,7 +142,7 @@ internal static ConfiguredCall Returns<T>(MatchArgs matchArgs, T returnThis, par
             }
             else
             {
-                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese));
+                returnValue = new ReturnMultipleValues<T>(new[] { returnThis }.Concat(returnThese).ToArray());
```

Comment:
Well, regarding performance, there are other bottlenecks we can address ;) For instance #248